app/services/repository_service.rb

```diff
@@ -0,0 +1,45 @@
+class RepositoryService
+
+  class InvalidExternalUrl < RuntimeError
+  end
+
+  def repository_by_id(repository_id)
+    Repository.find_by(id: repository_id)
+  end
+
+  def repository_by_url(url)
+    Repository.find_by(external_url: url)
+  end
+
+  def create_repository(product_service, url, attributes, is_custom_repository = false)
```

Comment:
This is the only thing meaningful this ServiceObject does. Please remove the other methods, and rename this service to `CreateRepositoryService` or something. Make sure it is the only public method and follow other best practices from [here](https://hackernoon.com/the-3-tenets-of-service-objects-c936b891b3c2).